experimental/segmenter/src/provider.rs

```diff
@@ -124,133 +40,32 @@ pub struct RuleBreakDataV1<'data> {
 }
 
 /// Property table for rule-based breaking.
-#[derive(Debug, PartialEq, Clone, yoke::Yokeable)]
+#[derive(Debug, PartialEq, Clone, yoke::Yokeable, zerofrom::ZeroFrom)]
 #[cfg_attr(
     feature = "provider_serde",
     derive(serde::Serialize, serde::Deserialize)
 )]
-pub enum RuleBreakPropertyTable<'data> {
-    #[cfg_attr(feature = "provider_serde", serde(borrow))]
-    Borrowed(&'data [[u8; 1024]; 897]),
-    Owned(Box<[[u8; 1024]; 897]>),
-}
-
-impl<'data> Deref for RuleBreakPropertyTable<'data> {
-    type Target = [[u8; 1024]; 897];
-    fn deref(&self) -> &Self::Target {
-        match self {
-            Self::Borrowed(v) => v,
-            Self::Owned(v) => v,
-        }
-    }
-}
-
-impl<'zf> zerofrom::ZeroFrom<'zf, RuleBreakPropertyTable<'_>> for RuleBreakPropertyTable<'zf> {
-    fn zero_from(cart: &'zf RuleBreakPropertyTable<'_>) -> Self {
-        RuleBreakPropertyTable::Borrowed(&*cart)
-    }
-}
+pub struct RuleBreakPropertyTable<'data>(
+    #[cfg_attr(feature = "provider_serde", serde(borrow))] pub ZeroVec<'data, u8>,
+);
 
```

Comment:
I see, so the data is `[[u8; 1024]; 897]` which is 918 KB for each of grapheme, sentence, and word.

Why is this data so much larger than the `[[u8; 1024]; 128]` for line break?

Reply:
Thanks for raising this question. Line segmenter encoding codepoints >= `U+20000` in https://github.com/unicode-org/icu4x/blob/b6c6994edc1ac8f069446063b5b5151a70c328eb/experimental/segmenter/src/line.rs#L216-L222 

but grapheme, sentence, and word break property define codepoints up to `U+E0FFF`, and they share the same underlying implementation, so we currently don't hard code them. 

I revisit the codepoint range of the uax 29 properties, and found a existing bug that we don't allocate enough space for grapheme cluster break. [154b49b](https://github.com/unicode-org/icu4x/pull/1652/commits/154b49b32b53887de9f54c986b57e63972f0def2) fixed this, but it makes the table a bit larger ... :(